### PR TITLE
fix macos sdk --- UNTESTED

### LINKED ISSUE #200: macos sdks out of date
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/main/recipe/download_osx_sdk.sh

this is out of date. this https://github.com/phracker/MacOSX-SDKs/releases only has up to 11.3

Options:
- rely on another repo (e.g. https://github.com/joseluisq/macosx-sdks/releases)
- force changing vmImage

### Installed packages

```shell
n.a
```

### Environment info

```shell
n.a
```

### PR DESCRIPTION
**Pull request by ngam**
fixes #200 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set name = "conda-forge-ci-setup" %}
-{% set version = "3.20.1" %}
+{% set version = "3.21.0" %}
```
Comment (ngam) [inline recipe/meta.yaml:2]: minor or patch release??

File: recipe/download_osx_sdk.sh [hunk 2]
```diff
@@ -22,9 +22,15 @@ export CONDA_BUILD_SYSROOT="${OSX_SDK_DIR}/MacOSX${MACOSX_SDK_VERSION}.sdk"
 
 if [[ ! -d ${CONDA_BUILD_SYSROOT} ]]; then
     echo "Downloading ${MACOSX_SDK_VERSION} sdk"
-    curl -L -O https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX${MACOSX_SDK_VERSION}.sdk.tar.xz
-    mkdir -p "$(dirname "$CONDA_BUILD_SYSROOT")"
-    tar -xf MacOSX${MACOSX_SDK_VERSION}.sdk.tar.xz -C "$(dirname "$CONDA_BUILD_SYSROOT")"
+    if [[ ${MACOSX_SDK_VERSION} == 12.* ]]; then
```
Comment (ngam) [inline recipe/download_osx_sdk.sh:25]: Does this logic even work?

Comment (hmaarrfk) [reply recipe/download_osx_sdk.sh:25]: maybe. you can put it in quotes too.

**Comment by hmaarrfk** [comment]
Are you able to upload this to your channel and then depend on your channel in addition to conda forges for some builds?

**Comment by ngam** [comment]
> Are you able to upload this to your channel and then depend on your channel in addition to conda forges for some builds?

I should first test it like this, right? I will try to do that

File: recipe/download_osx_sdk.sh (hunk 2 above)
Comment (ngam) [reply recipe/download_osx_sdk.sh:25]: we are only missing three of them btw, 12.0, 12.1, and 12.3

**Comment by ngam** [comment]
It will be good to hear from people about this in general. The vmImages from Azure do provide a selection of SDKs, and they're (I think) are organized based on support. So macos-11 will get you a number of them (quite a few), and macos-12 will get you the three missing ones. 

The problem in my pytorch PR is that I am trying to do it hands-off and the build is getting confused (i.e. told by macos-12 to set up sdk12.3 but told by conda-forge to set up sdk11.0, so the headers, etc are all over the place --- I could tidy it up there, but it would be good to pursue a unified strategy)

**Pull request closed without merging**